### PR TITLE
Fix django-filters in Juniper

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
**Description and justification**

This PR fixes users'/course enrollment/ generated certificates/proctored exams filter error, caused by django-filter forward compatibility issue. Also, makes it backward compatible.

The main issue that we have here is that in django-filter v1.0.4 -version used in Ironwood- there's a keyword `name` that was deprecated after v2.0.0 -Juniper uses v2.2.0-, then or we use the deprecated keyword `name` or we use the replacement `field_name`. 

Solution: 

Change name to field_name and change Django-filter to a version != than 1.0.4 when using python 2.7. Version 1.1.0 supports name and field_name keyword: https://github.com/carltongibson/django-filter/blob/1.1.0/django_filters/filters.py#L92

### CODE REVIEW

File: eox_core/api/data/v1/filters.py [hunk 1]
```diff
@@ -10,6 +12,43 @@
 from eox_core.edxapp_wrapper.users import get_course_enrollment
 
 
+def check_python_version():
```
Comment (morenol) [inline eox_core/api/data/v1/filters.py:15]: I think that we should use six for this.

See: https://six.readthedocs.io/#six.PY2

File: eox_core/settings/common.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def plugin_settings(settings):
     Defines eox-core settings when app is used as a plugin to edx-platform.
     See: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/plugins/README.rst
     """
-    settings.EOX_CORE_USERS_BACKEND = "eox_core.edxapp_wrapper.backends.users_j_v1"
+    settings.EOX_CORE_USERS_BACKEND = "eox_core.edxapp_wrapper.backends.users_h_v1"
```
Comment (andrey-canon) [inline eox_core/settings/common.py:22]: revert this please

File: eox_core/api/data/v1/filters.py [hunk 3]
```diff
@@ -199,4 +199,4 @@ class Meta(object):
             'course_id',
             'exam_name'
 
-        ]
+        ]
```
Comment (andrey-canon) [inline eox_core/api/data/v1/filters.py:202]: a line is missing

**Review by andrey-canon — COMMENTED** [review]
You have to force the django-filter update, maybe writing django-filter>1.0.4 in the base.in is enough

File: eox_core/api/data/v1/filters.py (hunk 1 above)
Comment (mariajgrimaldi) [reply eox_core/api/data/v1/filters.py]: thanks but I changed the solution!

Following what @andrey-canon suggested:

Given the problem I described, we could update django-filters in Ironwood from 1.0.4 to 1.1.0, this particular version supports name (raises just a deprecation warning) and field_name [reference](https://github.com/carltongibson/django-filter/blob/1.1.0/django_filters/filters.py#L92) What do you think?

File: requirements/base.txt [hunk 4]
```diff
@@ -11,7 +11,7 @@ celery==3.1.25            # via -c requirements/constraints.txt, -r requirements
 certifi==2020.6.20        # via requests
 chardet==3.0.4            # via requests
 django-crum==0.7.7        # via edx-proctoring
-django-filter==1.0.4      # via -c requirements/constraints.txt, -r requirements/base.in
+django-filter==1.1.0      # via -c requirements/constraints.txt, -r requirements/base.in
```
Comment (andrey-canon) [inline requirements/base.txt:14]: please use the command `make upgrade`  to update this file

Comment (mariajgrimaldi) [reply requirements/base.txt:14]: with python 3.5 or 2.7?

Comment (andrey-canon) [reply requirements/base.txt:14]: 2.7 maybe it's necessary the constraint to 1.1.0

**Review by andrey-canon — APPROVED** [review]
LGTM

**Pull request merged**